tests/unittests/Foundation/NSDecimalNumberTests.mm

```diff
@@ -0,0 +1,332 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <TestFramework.h>
+#import <Foundation/Foundation.h>
+#import <cmath>
+
+#define EXPECT_ARRAY_EQUAL(a, b)                     \
+    do {                                             \
+        for (int i = 0; i < NSDecimalMaxSize; ++i) { \
+            EXPECT_EQ(a[i], b[i]);                   \
+        }                                            \
+    } while (0, 0)
+
+static void testSubtract(NSDecimalNumber* x, NSDecimalNumber* y, NSDecimalNumber* expected) {
+    NSDecimalNumber* result = [x decimalNumberBySubtracting:y];
+    EXPECT_OBJCEQ(expected, result);
+}
+
+static void testAddition(NSDecimalNumber* x, NSDecimalNumber* y, NSDecimalNumber* expected) {
+    NSDecimalNumber* result = [x decimalNumberByAdding:y];
+    EXPECT_OBJCEQ(expected, result);
+}
+
+#define VALIDATE(decimal, len, exp, isNegative, mantissa) \
+    EXPECT_EQ(len, decimal._length);                      \
+    EXPECT_EQ(exp, decimal._exponent);                    \
+    EXPECT_EQ(isNegative, decimal._isNegative);           \
+    EXPECT_ARRAY_EQUAL(mantissa, decimal._mantissa);
```

Comment:
it doesn't help us much with the error message when we have failures. 

---
In reply to: [80738231](https://github.com/Microsoft/WinObjC/pull/1032#discussion_r80738231) [](ancestors = 80738231)